rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6802,115 +6804,245 @@ void getQueuedPegoutsCount_after_RSKIP271_activation() throws IOException {
+            bridgeConstantsRegtest.getErpFedActivationDelay(),
+            postRSKIP385_activations
         );
 
-        when(provider.getNewFederation()).thenReturn(erpFederation);
-
-        BridgeSupport bridgeSupport = bridgeSupportBuilder
-            .withActivations(activations)
-            .withBridgeConstants(bridgeConstantsRegtest)
-            .withProvider(provider)
-            .build();
+        Stream<Arguments> postRskip385 = Stream.of(
+            // active fed is standard and pegoutRequestsCount is equal to zero
+            Arguments.of(
+                postRSKIP385_activations,
+                standardFed,
+                0,
+                Coin.valueOf(65400L)
+            ),
+            // active fed is standard and pegoutRequestsCount is equal to one
+            Arguments.of(
+                postRSKIP385_activations,
+                standardFed,
+                1,
+                Coin.valueOf(68600L)
+            ),
+            // active fed is standard and there are many pegout requests
+            Arguments.of(
+                postRSKIP385_activations,
+                standardFed,
+                150,
+                Coin.valueOf(545400L)
+            ),
+            // active fed is p2sh and there are zero pegout requests
+            Arguments.of(
+                postRSKIP385_activations,
+                p2shFed,
+                0,
+                Coin.valueOf(158000L)
+            ),
+            // active fed is p2sh and there is one pegout request
+            Arguments.of(
+                postRSKIP385_activations,
+                p2shFed,
+                1,
+                Coin.valueOf(161200L)
+            ),
+            // active fed is p2sh and there are many pegout requests
+            Arguments.of(
+                postRSKIP385_activations,
+                p2shFed,
+                150,
+                Coin.valueOf(638000L)
+            )
+        );
+        return postRskip385;
+    }
 
-        int outputs = pegoutRequestsCount + 2; // N + 2 outputs
-        int pegoutTxSize = BridgeUtils.calculatePegoutTxSize(activations, erpFederation, 2, outputs);
+    private static Stream<Arguments> getEstimatedFeesForNextPegOutEventArgsProvider() {
+        BridgeRegTestConstants bridgeConstantsRegtest = BridgeRegTestConstants.getInstance();
 
-        Coin expected = feePerKB.multiply(pegoutTxSize).divide(1000);
+        Stream<Arguments> preRskip271 = getEstimatedFeesForNextPegOutEventArgsProvider_pre_RSKIP271(bridgeConstantsRegtest);
+        Stream<Arguments> preRskip385 = getEstimatedFeesForNextPegOutEventArgsProvider_pre_RSKIP385(bridgeConstantsRegtest);
+        Stream<Arguments> postRskip385 = getEstimatedFeesForNextPegOutEventArgsProvider_post_RSKIP385(bridgeConstantsRegtest);
 
-        assertEquals(expected, bridgeSupport.getEstimatedFeesForNextPegOutEvent());
+        return Stream.of(preRskip271, preRskip385, postRskip385).flatMap(Function.identity());
```

Comment:
Wouldn't it make more sense to use `concat`? This looks confusing, applying the identity function to each element, looks weird